pkg/driver/driver_azure.go

```diff
@@ -795,6 +811,42 @@ func (clients *azureDriverClients) deleteVM(ctx context.Context, resourceGroupNa
 	return nil
 }
 
+func (clients *azureDriverClients) checkOrphanNics(ctx context.Context, resourceGroupName, vmName string) error {
+	klog.V(2).Infof("Check for NIC leftovers belonging to deleted machine %q", vmName)
+	nicResults, err := clients.nic.List(ctx, resourceGroupName)
+	if err != nil {
+		return err
+	}
+	for _, nic := range nicResults.Values() {
+		if nic.Name == nil || !strings.Contains(*nic.Name, vmName) {
+			continue
+		}
+		klog.V(2).Infof("Found orphan NIC %q belonging to deleted machine %q", *nic.Name, vmName)
+		if err := clients.deleteNIC(ctx, resourceGroupName, *nic.Name); err != nil {
```

Comment:
Should we ignore the `NotFound` error just in case the resource was deleted after the `List` call but before `Delete` call could be made.

Reply:
Hmm should we do that already in the `deleteNIC` function? We could check if the future status is succeeded and if the status code == 404 and then return no error?
Same for `deleteDisk` funtion.
WDYT?

Reply:
yes. Sounds good. Better to be handled at the deepest level.

Reply:
I just saw we have already a `notFound` function to determine if an error is a 404. I simply used this one.